include/ada/url_aggregator-inl.h

```diff
@@ -11,14 +11,19 @@
 
 namespace ada {
 
-void url_aggregator::update_base_hash(const std::string_view input) {
+void url_aggregator::update_base_hash(std::optional<std::string> input) {
```

Comment:
The new function definition looks heavier and less performant.

Reply:
This is because we have to set it to nullopt in a case. We can have a different function to set it null if that makes more sense?